AMM/contracts/exchange/tests/functions/add_liquidity.rs

```diff
@@ -0,0 +1,363 @@
+use crate::utils::{
+    abi_calls::{add_liquidity, pool_info},
+    test_helpers::{
+        contract_balances, deposit_and_add_liquidity, deposit_but_do_not_add_liquidity, setup,
+        setup_and_initialize, setup_initialize_and_deposit_but_do_not_add_liquidity,
+        setup_initialize_deposit_and_add_liquidity, wallet_balances,
+    },
+    MetaAmounts,
+};
+use fuels::prelude::*;
+
+mod success {
+    use super::*;
+
+    #[tokio::test]
+    async fn adds_when_liquidity_is_zero() {
+        let (exchange, wallet, amounts, _asset_c_id) = setup_and_initialize().await;
+
+        let initial_pool_info = pool_info(&exchange.instance).await.value;
+        let initial_wallet_balances = wallet_balances(&exchange, &wallet).await;
+        let initial_contract_balances = contract_balances(&exchange).await;
+
+        deposit_but_do_not_add_liquidity(&amounts, &exchange).await;
+
+        let contract_balances_after_deposit = contract_balances(&exchange).await;
+
+        let added_liquidity = add_liquidity(
+            &exchange.instance,
+            CallParameters::default(),
+            TxParameters::default(),
+            amounts.liquidity,
+            amounts.deadline,
+        )
+        .await
+        .value;
+
+        let pool_info_after_adding_liquidity = pool_info(&exchange.instance).await.value;
```

Comment:
When doing a "before and after" check it should be fine to indicate which values are the initial values however the post-test values don't necessarily require that so you could name them like you would normally e.g. `pool_info`, `wallet_balance` etc. I think the implication is that there is 1 set of initial values and then the next appearance is for the post conditions. It shortens the names significantly. You may also use different naming conventions such as `post_<variable name>` e.g. `post_pool_info`. It's not great but if needed at least it presents some contrast.